okhttp/src/main/kotlin/okhttp3/OkHttpClient.kt

```diff
@@ -1066,6 +1096,29 @@ open class OkHttpClient internal constructor(
       this.minWebSocketMessageToCompress = bytes
     }
 
+    /**
+     * Set an override for host to allow insecure connections, typically a devserver with
```

Comment:
Here’s documentation that I think explains what we need to explain here:

> Configures OkHttp to not authenticate the HTTPS server on to [hostName]. This makes the client vulnerable to man-in-the-middle attacks and should only be used only in private development environments and only to carry test data.
>
> The following protections are removed:
>
> * The server’s TLS certificate **does not need to be signed** by a trusted certificate authority. Instead, OkHttp will trust any well-formed certificate, even if it is self-signed. This is necessary for testing against localhost, or in development environments where a certificate authority is not possible.
> * The server’s TLS certificate **does not need to match** the requested hostname. For example, if the certificate is issued to `example.com` and the request is to `localhost`, the certificate will still be accepted!
>
> Other TLS features are still used but provide no security benefits in absence of the above gaps. For example, an insecure TLS connection is capable of negotiating HTTP/2 with ALPN and it also has a regular-looking handshake.

Dig?

Reply:
There’s consequences here for our connection pools. We’ll need a test to confirm that the custom SSL objects implement enough equals and hashCode stuff to permit their connections to be pooled, and also enough to prevent their connections from being shared with OkHttpClient instances that don’t have these options.

Reply:
+1 I'll add a test